.github/actions/learning-path-staleness-check/index.js

```diff
@@ -0,0 +1,288 @@
+const prevPathPrefix = "prev/";
+const linePrefix = "#L";
+const separator = " | ";
+
+modifiedFilesDict = {};
+modifiedFilesUrlToFileName = {};
+
+var outOfSync = new Set();
+var manuallyReview = new Set();
+var suggestions = new Set();
+var modifiedFiles = new Set();
+
+const oldNewLinkSeparator = ' -> ';
+
+let modifiedFilesToCommit = [];
+
+function AppendModifiedFilesToCommit(path, core)
+{
+  modifiedFilesToCommit.push(path)
+  core.setOutput('modifiedFilesToCommit', modifiedFilesToCommit.join(' '))
+}
+
+function ReplaceOldWithNewText(content, oldText, newText)
+{
+  return content.replaceAll(oldText, newText);
+}
+
+function UpdateModifiedFiles(fileName, path, learningPathFile)
+{
+  modifiedFilesUrlToFileName[path] = fileName;
+
+  modifiedFilesDict[path] = modifiedFilesDict[path] ? modifiedFilesDict[path] : new Set();;
+  modifiedFilesDict[path].add(learningPathFile);
+
+  modifiedFiles = new Set();
+  for (currPath in modifiedFilesDict)
+  {
+    const fileName = modifiedFilesUrlToFileName[currPath];
+    modifiedFiles.add(AssembleModifiedFilesOutput(fileName, currPath, Array.from(modifiedFilesDict[currPath])));
+  }
+}
+
+function AssembleModifiedFilesOutput(fileName, path, learningPathFiles)
+{
+  return CreateLink(fileName, path, undefined) + separator + BoldedText(learningPathFiles.join(" "));
+}
+
+function BoldedText(text)
+{
+  return "**" + text + "**";
+}
+
+function UpdateManuallyReview(fileName, path, learningPathFile, learningPathLineNumber, lineNumber = undefined)
+{
+  manuallyReview.add(AssembleOutput(fileName, path, undefined, lineNumber, undefined, learningPathFile, learningPathLineNumber))
+}
+
+function UpdateOutOfSync(link, learningPathFile)
+{
+  outOfSync.add(link + separator + BoldedText(learningPathFile))
+}
+
+function UpdateSuggestions(fileName, oldPath, newPath, learningPathFile, learningPathLineNumber, oldLineNumber, newLineNumber)
+{
+  suggestions.add(AssembleOutput(fileName, oldPath, newPath, oldLineNumber, newLineNumber, learningPathFile, learningPathLineNumber))
+}
+
+function CreateLink(fileName, path, lineNumber)
+{
+  var codeFileLink = "[" + fileName + "]" + "(" + path + ")"
+  return AppendLineNumber(codeFileLink, lineNumber)
+}
+
+function AssembleOutput(fileName, oldPath, newPath, oldLineNumber, newLineNumber, learningPathFile, learningPathLineNumber)
+{
+  var codeFileLink = CreateLink(fileName, oldPath, oldLineNumber)
+
+  if (newPath && newLineNumber) {
+    codeFileLink += oldNewLinkSeparator + CreateLink(fileName, newPath, newLineNumber)
+  }
+
+  return codeFileLink + separator + BoldedText(AppendLineNumber(learningPathFile, learningPathLineNumber, undefined));
+}
+
+function AppendLineNumber(text, lineNumber)
+{
+  if (!lineNumber) { return text }
+
+  return text + " " + linePrefix + lineNumber
+}
+
+function CheckForEndOfLink(str, startIndex)
+{
+  const illegalCharIndex = str.substr(startIndex).search("[(), '\`\"\}\{]|\. "); // This regex isn't perfect, but should cover most cases.
```

Comment:
Do you mind providing some examples of what links you expect to handle here? This regex doesn't look correct.

E.g. Are you only handles markdown links for now? `[friendly text](URL)`, or are you trying to handle any link that appears in the file, regardless of it its annotated?

Reply:
To expand on this, ignoring the correctness of the regex, depending on the examples of urls you are trying to handle this regex likely is not necessary in the first place

Reply:
Though in the case of our learning docs I believe basically everything does follow the `[]()` format, this should also handle normal links in text with a trailing space (I also made an adjustment to look for newlines after periods in the latest version)